### PR TITLE
[VSCODE] Extension for Wake Syntax

### PR DESCRIPTION
**Pull request by azidar**
Added @terpstra's source code for wake syntax, with VSCode extension details. To use you can open `share/doc/wake/syntax/vscode` in VSCode, then press F5 to open a new VSCode window with the wake extension installed. Opening other projects with that window should have wake syntax highlighting working.

### CODE REVIEW

**Comment by mwachs5** [comment]
can/should I try this out?

**Comment by azidar** [comment]
Sure!

**Comment by mwachs5** [comment]
This didn't work for me, but also pressing F5 is non trivial on most modern keyboards... is there something more verbose to do?

File: share/doc/wake/syntax/vscode/language-configuration.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    // symbols used as brackets
+    "brackets": [
+        ["{", "}"],
+        ["[", "]"],
+        ["(", ")"]
+    ],
+    // symbols that are auto closed when typing
+    "autoClosingPairs": [
+        ["{", "}"],
+        ["[", "]"],
```
Comment (terpstra) [inline share/doc/wake/syntax/vscode/language-configuration.json:15]: future proofing?

File: share/doc/wake/syntax/vscode/syntaxes/wake.tmLanguage.json [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+			]
+		},
+		"parens": {
+			"name": "expression.group",
+			"begin": "\\(",
+			"end": "\\)",
+			"beginCaptures": { "0": { "name": "keyword.operator.wake" } },
+			"endCaptures": { "0": { "name": "keyword.operator.wake" } },
+			"patterns": [ { "include": "#expression" } ]
+		},
```
Comment (terpstra) [inline share/doc/wake/syntax/vscode/syntaxes/wake.tmLanguage.json:28]: Tried to make a real parser, but wake indentation is context sensitive => cannot be done with a simple parser.

File: share/doc/wake/syntax/vscode/syntaxes/wake.tmLanguage.json [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+			}]
+		},
+		"operator": {
+			"name":	"keyword.operator.wake",
+			"match": "[~!$%^&*+=\\\\|;<,.>/-]+"
+		},
+		"identifier": {
+			"name": "meta.identifier.wake",
+			"match": "[A-Za-z_][A-Za-z_0-9]*"
+		},
```
Comment (terpstra) [inline share/doc/wake/syntax/vscode/syntaxes/wake.tmLanguage.json:42]: Operators and identifiers ignore unicode. Probably fine for now.

**Review by terpstra — APPROVED** [review]

**Pull request merged**